sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalStructureSameImplementation.cs

```diff
@@ -67,42 +73,50 @@ protected sealed override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var switchSection = (SwitchSectionSyntax) c.Node;
+                    var switchSection = (SwitchSectionSyntax)c.Node;
+
+                    if (GetStatements(switchSection).Where(IsNotIgnored).AreLessThan(2))
+                    {
+                        return;
+                    }
+
                     var precedingSection = switchSection
                         .GetPrecedingSections()
                         .FirstOrDefault(
                             preceding => EquivalenceChecker.AreEquivalent(switchSection.Statements, preceding.Statements));
 
                     if (precedingSection != null)
                     {
-                        ReportSection(c, switchSection, precedingSection);
+                        ReportSyntaxNode(c, switchSection, precedingSection, "case");
                     }
                 },
                 SyntaxKind.SwitchSection);
         }
 
-        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statementToCheck,
+        private static IEnumerable<StatementSyntax> GetStatements(SwitchSectionSyntax switchSection)
+        {
+            return Enumerable.Empty<StatementSyntax>()
+                .Union(switchSection.Statements.OfType<BlockSyntax>().SelectMany(block => block.Statements))
+                .Union(switchSection.Statements.Where(s => !s.IsKind(SyntaxKind.Block)));
+        }
+
+        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statement,
             IEnumerable<StatementSyntax> precedingStatements)
         {
+            if (!statement.ChildNodes().AreMoreThan(1))

```

Comment:
I would negate the condition and would probably extract it into some variable before the `if` condition.